# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  1264ad385430474baddd19af677999ecd829a1c6
Misc: Make sure that archive folder has Archive flag.

diff --git a/mail/modules/MessageArchiver.sys.mjs b/mail/modules/MessageArchiver.sys.mjs
--- a/mail/modules/MessageArchiver.sys.mjs
+++ b/mail/modules/MessageArchiver.sys.mjs
@@ -150,8 +150,13 @@ MessageArchiver.prototype = {
         archiveKeepFolderStructure = identity.archiveKeepFolderStructure;
         archiveRecreateInbox = identity.archiveRecreateInbox;
       }
 
+      if (!archiveFolder.getFlag(Ci.nsMsgFolderFlags.Archive)) {
+        console.info(`Setting Archive flag on folder ${archiveFolder.URI}`);
+        archiveFolder.setFlag(Ci.nsMsgFolderFlags.Archive);
+      }
+
       let copyBatchKey = msgHdr.folder.URI;
       if (archiveGranularity >= Ci.nsIMsgIdentity.perYearArchiveFolders) {
         copyBatchKey += "\0" + msgYear;
       }
